lib/history/sqlite/sqlite.go

```diff
@@ -30,115 +29,142 @@ import (
 
 	"github.com/gravitational/trace"
 	"github.com/jmoiron/sqlx"
-	"github.com/jonboulle/clockwork"
-	_ "github.com/mattn/go-sqlite3" // initialize sqlite3
+	"github.com/jonboulle/clockwork" // initialize sqlite3
 	log "github.com/sirupsen/logrus"
 )
 
-// Timeline represents a timeline of cluster status events. The timeline
-// can hold a specified amount of events and uses a FIFO eviction policy.
+// Timeline represents a timeline of status events.
 // Timeline events are stored in a local sqlite database.
+// The timeline will retain events for a specified duration and then deleted.
 //
 // Implements history.Timeline
 type Timeline struct {
 	// Config contains timeline configuration.
-	Config Config
-	// size specifies the current number of events stored in the timeline.
-	size int
+	config Config
 	// database points to underlying sqlite database.
 	database *sqlx.DB
-	// lastStatus holds the last recorded cluster status.
-	lastStatus *pb.SystemStatus
-	// mu locks timeline access.
-	mu sync.Mutex
+	// lastStatus holds the last recorded status.
+	lastStatus *pb.NodeStatus
+	// statusMutex locks access to the lastStatus field.
+	statusMutex sync.Mutex
 }
 
 // Config defines Timeline configuration.
 type Config struct {
 	// DBPath specifies the database location.
 	DBPath string
-	// Capacity specifies the max number of events that can be stored in the timeline.
-	Capacity int
+	// RetentionDuration specifies the duration to store events.
+	RetentionDuration time.Duration
 	// Clock will be used to record event timestamps.
 	Clock clockwork.Clock
 }
 
 // NewTimeline initializes and returns a new Timeline with the
 // specified configuration.
 func NewTimeline(ctx context.Context, config Config) (*Timeline, error) {
-	database, err := initSQLite(ctx, config.DBPath)
-	if err != nil {
-		return nil, trace.Wrap(err)
+	timeline := &Timeline{
+		config: config,
 	}
 
-	size, err := getSize(ctx, database)
-	if err != nil {
-		database.Close()
-		return nil, trace.Wrap(err)
+	if err := timeline.initSQLite(ctx); err != nil {
+		return nil, trace.Wrap(err, "failed to init sqlite")
 	}
 
-	return &Timeline{
-		Config:   config,
-		size:     size,
-		database: database,
-		// TODO: store and recover lastStatus in case satellite agent restarts.
-		lastStatus: nil,
-	}, nil
+	// if err := timeline.initPrevStatus(ctx); err != nil {
+	// 	return nil, trace.Wrap(err, "failed to init previous status")
+	// }
+
+	// TODO: should this be called in constructor method?
```

Comment:
I think it's fine unless the caller can have a reason to instantiate the timeline multiple times?